src/streamlink_cli/main.py

```diff
@@ -407,7 +407,8 @@ def handle_stream(plugin, streams, stream_name):
 
     # Print JSON representation of the stream
     elif console.json:
-        console.msg_json(stream)
+        title = create_title(plugin)
```

Comment:
`create_title` is no more. This has just been replaced in #3962.

Unfortunately I had this PR here off the radar and moved the new title/filename formatter initialization to a different method which won't work for these changes. The formatter (see down below) needs to be moved to the `handle_url` method and passed as an argument to `handle_stream`, so both `console.msg_json()` calls can read the title from the formatter instance via

```py
formatter.title(args.title, defaults=DEFAULT_STREAM_METADATA) if args.title else args.url
```